openmls/src/schedule/psk.rs

```diff
@@ -458,3 +464,83 @@ impl From<Secret> for PskSecret {
         Self { secret }
     }
 }
+
+pub(crate) fn load_psks<'p>(
+    key_store: &impl OpenMlsKeyStore,
+    resumption_psk_store: &ResumptionPskStore,
+    psk_ids: &'p [PreSharedKeyId],
+) -> Result<Vec<(&'p PreSharedKeyId, Secret)>, PskError> {
+    let mut psk_bundles = Vec::new();
+
+    for psk_id in psk_ids.iter() {
+        log_crypto!(trace, "PSK store {:?}", resumption_psk_store);
+
+        match &psk_id.psk {
+            Psk::Resumption(resumption) => {
+                if let Some(psk_bundle) = resumption_psk_store.get(resumption.psk_epoch()) {
+                    // TODO
+                    psk_bundles.push((psk_id, psk_bundle.secret.clone()));
+                } else {
+                    return Err(PskError::KeyNotFound);
+                }
+            }
+            Psk::External(_) => {
+                if let Some(psk_bundle) = key_store.read::<PskBundle>(&psk_id.keystore_id()?) {
+                    psk_bundles.push((psk_id, psk_bundle.secret));
+                } else {
+                    return Err(PskError::KeyNotFound);
+                }
+            }
+        }
+    }
+
+    Ok(psk_bundles)
+}
+
+// ----- ResumptionPskStore ------------------------------------------------------------------------
+
+/// Resumption psks store. This is where the resumption psks are kept in a
+/// rollover list.
+#[derive(Debug, Serialize, Deserialize)]
```

Comment:
As far as I can tell this code was only moved, so nothing to touch here, but this store needs to be updated.

I don't think that this should be serialised. This way we end up writing secrets to the disk in an unprotected way. But the secrets should be in the key store anyway such that there should be no need to serialise these within the group itself.

The behaviour of the `add` function is pretty difficult to understand for callers because they doesn't know whether they removed a value or not. You could solve this by returning the optional old value.

Reply:
I filed #1355 and #1356.